TombEngine/Game/control/box.cpp

```diff
@@ -137,77 +138,74 @@ bool MoveCreature3DPos(Pose* fromPose, Pose* toPose, int velocity, short angleDi
 	return false;
 }
 
-void CreatureYRot2(Pose* fromPose, short angle, short angleAdd) 
+void CreatureYRot2(Pose* fromPose, short angle, short angleAdd)
 {
-	if (angleAdd < angle)
+	if (angle > angleAdd)
 	{
 		fromPose->Orientation.y += angleAdd;
-		return;
-	} 
-
-	if (angle < -angleAdd)
+	}
+	else if (angle < -angleAdd)
 	{
 		fromPose->Orientation.y -= angleAdd;
-		return;
-	} 
-
-	fromPose->Orientation.y += angle;
+	}
+	else
+	{
+		fromPose->Orientation.y += angle;
+	}
 }
 
 bool SameZone(CreatureInfo* creature, ItemInfo* target)
 {
-	int* zone = g_Level.Zones[(int)creature->LOT.Zone][FlipStatus].data();
-	auto* item = &g_Level.Items[creature->ItemNumber];
+	auto& item = g_Level.Items[creature->ItemNumber];
+	auto* zone = g_Level.Zones[(int)creature->LOT.Zone][FlipStatus].data();
 
-	auto* room = &g_Level.Rooms[item->RoomNumber];
-	FloorInfo* floor = GetSector(room, item->Pose.Position.x - room->x, item->Pose.Position.z - room->z);
-	if (floor->Box == NO_BOX)
+	auto& roomSource = g_Level.Rooms[item.RoomNumber];
+	auto& boxSource = GetSector(&roomSource, item.Pose.Position.x - roomSource.x, item.Pose.Position.z - roomSource.z)->Box;
+	if (boxSource == NO_BOX)
 		return false;
+	item.BoxNumber = boxSource;
 
-	item->BoxNumber = floor->Box;
-
-	room = &g_Level.Rooms[target->RoomNumber];
-	floor = GetSector(room, target->Pose.Position.x - room->x, target->Pose.Position.z - room->z);
-
-	if (floor->Box == NO_BOX)
+	auto& roomTarget = g_Level.Rooms[target->RoomNumber];
+	auto& boxTarget = GetSector(&roomTarget, target->Pose.Position.x - roomTarget.x, target->Pose.Position.z - roomTarget.z)->Box;
+	if (boxTarget == NO_BOX)
 		return false;
+	target->BoxNumber = boxTarget;
 
-	target->BoxNumber = floor->Box;
-
-	return (zone[item->BoxNumber] == zone[target->BoxNumber]);
+	return (zone[item.BoxNumber] == zone[target->BoxNumber]);
 }
 
 short AIGuard(CreatureInfo* creature) 
 {
-	auto* item = &g_Level.Items[creature->ItemNumber];
-	if (item->AIBits & (GUARD | PATROL1))
+	auto& item = g_Level.Items[creature->ItemNumber];
+	if (item.AIBits & (MODIFY|PATROL1|PATROL2|AMBUSH|FOLLOW))
```

Comment:
I still don't understand why you have assigned all AI bits here. Original tomb4 code seems to use just `MODIFY` flag.

Reply:
it seem this AIGuard() should only use AI_GUARD ai only, don't know why it was just modify.

Reply:
Fixed